tkg/client/init.go

```diff
@@ -456,6 +456,30 @@ func (c *TkgClient) MoveObjects(fromKubeconfigPath, toKubeconfigPath, namespace
 	return c.clusterctlClient.Move(moveOptions)
 }
 
+// MoveTKRAndOSImages moves the resolved TKR and associated OSImage CR for a Cluster resource
+// from namespaceto a target management
+func (c *TkgClient) MoveTKRAndOSImages(fromClusterClient, toClusterClient clusterclient.Client) error {
+	tkr, err := fromClusterClient.GetClusterResolvedTanzuKubernetesRelease()
+	if tkr == nil {
+		return err // error occurs or management cluster does not have resolved TKR
+	}
+	osImages, err := fromClusterClient.GetClusterResolvedOSImages()
+	if err != nil {
+		return err
+	}
+	tkr.SetResourceVersion("")
```

Comment:
It is better to recreate tkr object with only copy name/namespace in metadata & whole tkr.spec?

Since it may have creation timestamp, annotation, labels and etc in metadata.

Reply:
this should be fine. checkout how CAPI performs the move [here](https://github.com/kubernetes-sigs/cluster-api/blob/1bf70d5c24004977844b9da7570853c32e29c6b3/cmd/clusterctl/client/cluster/mover.go#L865). it is getting the object from cluster API server like what we do here (instead of constructing one)